src/store/freesoundApi.ts

```diff
@@ -0,0 +1,50 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+
+export interface SoundEffect {
+  id: number;
+  name: string;
+  previews: {
+    'preview-hq-mp3': string;
+    'preview-hq-ogg': string;
+    'preview-lq-mp3': string;
+    'preview-lq-ogg': string;
+  };
+}
+
+export interface FreesoundSearchResponse {
+  count: number;
+  results: SoundEffect[];
+}
+
+interface SearchParams {
+  query: string;
+  page: number;
+}
+
+const FREESOUND_API_KEY = 'M80wfthcPGELZtZRI0vUKoU8aVEtwAum5raYonBg';
```

Comment:
As a rule of thumb, you should always avoid commiting API keys to the repository. This is what environment variables are for.

https://vite.dev/guide/env-and-mode